### PR TITLE
Separate ipfamilies

### PR DESCRIPTION
**Pull request by thetechnick**
Seperated IPv6 and IPv4, because it was stupid to combine them in the first place.

### CODE REVIEW

File: internal/controllers/iplease_controller.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+			Type:               ipamv1alpha1.IPLeaseBound,
+			Reason:             "Unavailable",
+			Message:            "No more IPs available from pool.",
+			ObservedGeneration: iplease.GetGeneration(),
+			Status:             metav1.ConditionFalse,
+		})
+		return ctrl.Result{
+			// Retry to allocate later.
+			RequeueAfter: 5 * time.Second,
+		}, nil
```
Comment (erdii) [inline internal/controllers/iplease_controller.go:200]: oops

```suggestion
		}, r.Status().Update(ctx, iplease.ClientObject())
```

Comment (erdii) [reply internal/controllers/iplease_controller.go:200]: LGTM :)

File: api/v1alpha1/ipv4lease_types.go [hunk 2]
```diff
@@ -19,75 +19,66 @@ package v1alpha1
...
+	Type IPLeaseType `json:"type"`
+
 	// Static IP lease settings.
-	Static *StaticIPLease `json:"static,omitempty"`
-	// IPFamilies that this lease should include.
-	IPFamilies []corev1.IPFamily `json:"ipFamilies"`
+	Static *IPLeaseStatic `json:"static,omitempty"`
+
 	// Renew time is the time when the lease holder has last updated the lease.
 	// Falls back to .metadata.creationTimestamp if not set.
```
Comment (erdii) [inline api/v1alpha1/ipv4lease_types.go:39]: This is not true for kubectl:

image

Comment (erdii) [reply api/v1alpha1/ipv4lease_types.go:39]: I think `.spec.renewTime` is unused and can be removed?

File: internal/controllers/iplease_controller.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	log := r.Log.WithValues("iplease", req.NamespacedName)
+
+	iplease := adapter.AdaptIPLease(
+		r.IPLeaseType.ClientObject().DeepCopyObject())
+	if err = r.Get(ctx, req.NamespacedName, iplease.ClientObject()); err != nil {
+		return res, client.IgnoreNotFound(err)
+	}
+	defer func() {
+		// ensure that no matter how we exit the reconcile function,
+		// we want to reconcile the IPLease after the lease duration expired.
```
Comment (erdii) [inline internal/controllers/iplease_controller.go:66]: Is it really needed to wait for lease expiration when a lease already has been deleted?

**Comment by thetechnick** [comment]
Fixes commited here:
https://github.com/routerd/routerd/pull/1/commits/ea4ccab836d94e8d19d68349b0c9e48e6523e1be

Closing this PR (and later the Repo together with kube-dhcp) in favor of putting everything under routed/routerd for now.

**Pull request closed without merging**